equinix/resource_metal_virtual_circuit.go

```diff
@@ -289,71 +320,117 @@ func getVCStateWaiter(client *packngo.Client, id string, timeout time.Duration,
+		if d.HasChange("speed") {
+			needsUpdate = true
+			speed := d.Get("speed").(string)
+			ur.VlanVirtualCircuitUpdateInput.Speed = &speed
+		}
+
+		if d.HasChange("tags") {
+			needsUpdate = true
+			ts := d.Get("tags")
+			sts := []string{}
+
+			switch ts.(type) {
+			case []interface{}:
+				for _, v := range ts.([]interface{}) {
+					sts = append(sts, v.(string))
+				}
+				ur.VlanVirtualCircuitUpdateInput.Tags = sts
+			default:
+				return diag.Errorf("garbage in tags: %s", ts)
+			}
+		}
+	} else {
+		ur.VrfVirtualCircuitUpdateInput = &metalv1.VrfVirtualCircuitUpdateInput{}
+
+		if d.HasChange("name") {
+			needsUpdate = true
+			name := d.Get("name").(string)
+			ur.VrfVirtualCircuitUpdateInput.Name = &name
+		}
+
+		if d.HasChange("description") {
+			needsUpdate = true
+			desc := d.Get("description").(string)
+			ur.VrfVirtualCircuitUpdateInput.Description = &desc
+		}
+
+		if d.HasChange("speed") {
+			needsUpdate = true
+			speed := d.Get("speed").(string)
+			ur.VrfVirtualCircuitUpdateInput.Speed = &speed
+		}
 
-		switch ts.(type) {
-		case []interface{}:
-			for _, v := range ts.([]interface{}) {
-				sts = append(sts, v.(string))
+		if d.HasChange("tags") {
+			needsUpdate = true
+			ts := d.Get("tags")
+			sts := []string{}
+
+			switch ts.(type) {
+			case []interface{}:
+				for _, v := range ts.([]interface{}) {
+					sts = append(sts, v.(string))
+				}
+				ur.VrfVirtualCircuitUpdateInput.Tags = sts
+			default:
+				return diag.Errorf("garbage in tags: %s", ts)
 			}
-			ur.Tags = &sts
-		default:
-			return diag.Errorf("garbage in tags: %s", ts)
 		}
 	}
 
-	if !reflect.DeepEqual(ur, packngo.VCUpdateRequest{}) {
-		if _, _, err := client.VirtualCircuits.Update(d.Id(), &ur, nil); err != nil {
+	if needsUpdate {
+		if _, _, err := client.InterconnectionsApi.UpdateVirtualCircuit(ctx, d.Id()).VirtualCircuitUpdateInput(ur).Execute(); err != nil {
 			return diag.FromErr(equinix_errors.FriendlyError(err))
 		}
 	}
 	return resourceMetalVirtualCircuitRead(ctx, d, meta)
 }
 
 func resourceMetalVirtualCircuitDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	meta.(*config.Config).AddModuleToMetalUserAgent(d)
-	client := meta.(*config.Config).Metal
+	client := meta.(*config.Config).NewMetalClientForSDK(d)
 
-	resp, err := client.VirtualCircuits.Delete(d.Id())
-	if equinix_errors.IgnoreResponseErrors(equinix_errors.HttpForbidden, equinix_errors.HttpNotFound)(resp, err) != nil {
+	_, resp, err := client.InterconnectionsApi.DeleteVirtualCircuit(ctx, d.Id()).Execute()
+	if equinix_errors.IgnoreHttpResponseErrors(equinix_errors.HttpForbidden, equinix_errors.HttpNotFound)(resp, err) != nil {
 		return diag.FromErr(equinix_errors.FriendlyError(err))
 	}
 
 	deleteWaiter := getVCStateWaiter(
+		ctx,
 		client,
 		d.Id(),
 		d.Timeout(schema.TimeoutDelete)-30*time.Second,
-		[]string{string(packngo.VCStatusDeleting)},
+		[]string{string(metalv1.VLANVIRTUALCIRCUITSTATUS_DELETING)},
```

Comment:
Status checks are using the values defined in the VLAN VC schema but that's probably OK for now since the previous SDK didn't even distinguish between VC types